src/nvim/mark_extended.h

```diff
@@ -0,0 +1,65 @@
+#ifndef NVIM_EXTMARK_H
+#define NVIM_EXTMARK_H
+
+#include "nvim/mark_extended_defs.h"
+#include "nvim/lib/kbtree.h"
+#include "nvim/lib/kvec.h"
+#include "nvim/map.h"
+
+typedef PMap(cstr_t) StringMap2; // TODO do these have to be seperate?
+
+typedef struct ExtendedMark ExtendedMark;
+struct ExtendedMark {
+  StringMap2 *names; // key = namespace, value = id
```

Comment:
By delething this i am requried to add a btree for each namespace. The current implementation of btree doesn't allow me to do this. Will see what i can do...